syntaxes/armv8-a.tmlanguage.json

```diff
@@ -0,0 +1,221 @@
mlsll|smlslt|smlsl|smmla|smnegl|smopa|smops|smov|smstart|smstop|smsubl|smulh|smull2|smullb|smullt|smull|splice|sqabs|sqadd|sqcadd|sqcvtn|sqcvtun|sqcvtu|sqcvt|sqdecb|sqdecd|sqdech|sqdecp|sqdecw|sqdmlal2|sqdmlalbt|sqdmlalb|sqdmlalt|sqdmlal|sqdmlsl2|sqdmlslbt|sqdmlslb|sqdmlslt|sqdmlsl|sqdmulh|sqdmull2|sqdmullb|sqdmullt|sqdmull|sqincb|sqincd|sqinch|sqincp|sqincw|sqneg|sqrdcmlah|sqrdmlah|sqrdmlsh|sqrdmulh|sqrshlr|sqrshl|sqrshrn2|sqrshrnb|sqrshrnt|sqrshrn|sqrshrun2|sqrshrunb|sqrshrunt|sqrshrun|sqrshru|sqrshr|sqshlr|sqshlu|sqshl|sqshrn2|sqshrnb|sqshrnt|sqshrn|sqshrun2|sqshrunb|sqshrunt|sqshrun|sqsubr|sqsub|sqxtn2|sqxtnb|sqxtnt|sqxtn|sqxtun2|sqxtunb|sqxtunt|sqxtun|srhadd|sri|srshlr|srshl|srshr|srsra|ssbb|sshll2|sshllb|sshllt|sshll|sshl|sshr|ssra|ssubl2|ssublbt|ssublb|ssubltb|ssublt|ssubl|ssubw2|ssubwb|ssubwt|ssubw|st1b|st1d|st1h|st1q|st1w|st1|st2b|st2d|st2g|st2h|st2q|st2w|st2|st3b|st3d|st3h|st3q|st3w|st3|st4b|st4d|st4h|st4q|st4w|st4|st64bv0|st64bv|st64b|staddb|staddh|staddlb|staddlh|staddl|stadd|stclrb|stclrh|stclrlb|stclrlh|stclrl|stclr|steorb|steorh|steorlb|steorlh|steorl|steor|stgm|stgp|stg|stilp|stl1|stllrb|stllrh|stllr|stlrb|stlrh|stlr|stlurb|stlurh|stlur|stlxp|stlxrb|stlxrh|stlxr|stnp|stnt1b|stnt1d|stnt1h|stnt1w|stp|strb|strh|str|stsetb|stseth|stsetlb|stsetlh|stsetl|stset|stsmaxb|stsmaxh|stsmaxlb|stsmaxlh|stsmaxl|stsmax|stsminb|stsminh|stsminlb|stsminlh|stsminl|stsmin|sttrb|sttrh|sttr|stumaxb|stumaxh|stumaxlb|stumaxlh|stumaxl|stumax|stuminb|stuminh|stuminlb|stuminlh|stuminl|stumin|sturb|sturh|stur|stxp|stxrb|stxrh|stxr|stz2g|stzgm|stzg|subg|subhn2|subhnb|subhnt|subhn|subps|subp|subr|subs|sub|sudot|sumlall|sumopa|sumops|sunpkhi|sunpklo|sunpk|suqadd|suvdot|svc|svdot|swpab|swpah|swpalb|swpalh|swpal|swpa|swpb|swph|swplb|swplh|swpl|swppal|swppa|swppl|swpp|swp|sxtb|sxth|sxtl2|sxtl|sxtw|sysl|sysp|sys|tblq|tbl|tbnz|tbxq|tbx|tbz|tcancel|tcommit|tlbip|tlbi|trcit|trn1|trn2|tsbcsync|tstart|tst|ttest|uabal2|uabalb|uabalt|uabal|uaba|uabdl2|uabdlb|uabdlt|uabdl|uabd|uadalp|uaddl2|uaddlb|uaddlp|uaddlt|uaddlv|uaddl|uaddv|uaddw2|uaddwb|uaddwt|uaddw|ubfiz|ubfm|ubfx|uclamp|ucvtf|udf|udivr|udiv|udot|uhadd|uhsubr|uhsub|umaddl|umaxp|umaxqv|umaxv|umax|uminp|uminqv|uminv|umin|umlal2|umlalb|umlall|umlalt|umlal|umlsl2|umlslb|umlsll|umlslt|umlsl|ummla|umnegl|umopa|umops|umov|umsubl|umulh|umull2|umullb|umullt|umull|uqadd|uqcvtn|uqcvt|uqdecb|uqdecd|uqdech|uqdecp|uqdecw|uqincb|uqincd|uqinch|uqincp|uqincw|uqrshlr|uqrshl|uqrshrn2|uqrshrnb|uqrshrnt|uqrshrn|uqrshr|uqshlr|uqshl|uqshrn2|uqshrnb|uqshrnt|uqshrn|uqsubr|uqsub|uqxtn2|uqxtnb|uqxtnt|uqxtn|urecpe|urhadd|urshlr|urshl|urshr|ursqrte|ursra|usdot|ushll2|ushllb|ushllt|ushll|ushl|ushr|usmlall|usmmla|usmopa|usmops|usqadd|usra|usubl2|usublb|usublt|usubl|usubw2|usubwb|usubwt|usubw|usvdot|uunpkhi|uunpklo|uunpk|uvdot|uxtb|uxth|uxtl2|uxtl|uxtw|uzp1|uzp2|uzpq1|uzpq2|uzp|whilege|whilegt|whilehi|whilehs|whilele|whilelo|whilels|whilelt|whilerw|whilewr|wrffr|xaflag|xar|xpacd|xpaci|xpaclri|xtn2|xtn|yield|zero|zip1|zip2|zipq1|zipq2|zip)(?-i)\\b",
```

Comment:
```sh
curl --silent 'https://raw.githubusercontent.com/compiler-explorer/compiler-explorer/730437464d7900a1f232345db0bc196bfa1bbaeb/lib/asm-docs/generated/asm-docs-aarch64.ts' | grep '        case "' | cut -f 2 -d '"' | grep -v '<cc>' | grep -v '<r>' | tr '[:upper:]' '[:lower:]' | sort | uniq | while IFS= read -r line; do
    echo "${line}ū"
done | sort | sed 's/ū$//'
```

the adding of the `ū` is to get "absence of character" to sort after a character, i.e. `zip1|zip2|zipq1|zipq2|zip` instead of `zip|zip1|zip2|zipq1|zipq2` because otherwise the regex will match the shorter one first. Which is an issue for some instructions actually in the currently existing ARM grammar like for example in this line:

```json
      "match": "\\b(?i)(?:ldar[bh]?|ldax[pb(?:rh?)]|ldn?p|ldpsw|ldr[bh]?|ldrsb|ldrsh|ldrsw|ldtr[bh]?|ldtrsw|ldurb?|ldursb|ldursh|ldursw|ldx[pr]|ldxr[bh])(?-i)\\b",
```
it should be
```json
      "match": "\\b(?i)(?:ldar[bh]?|ldax[pb(?:rh?)]|ldn?p|ldpsw|ldrsb|ldrsh|ldrsw|ldr[bh]?|ldtr[bh]?|ldtrsw|ldurb?|ldursb|ldursh|ldursw|ldx[pr]|ldxr[bh])(?-i)\\b",
```

otherwise none of the instructions after `ldr[bh]?`, namely `ldpsw|ldrsb|ldrsh|ldrsw` will match, since `ldr` will match first.

Reply:
I put all the instructions in one place because I don't have enough knowledge of assembly to group 1523 instructions and because I didn't really get why you were grouping them and highlighting errors. Ideally, we would would parse the xml files in the documentation (which include the grammar of each instruction) and generate this code.